dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentletLoader.java

```diff
@@ -0,0 +1,851 @@
+            }
+        } else {
+            sb.append("#set($isWidget= \"")
+                .append(false)
+                .append("\")");
+        }
+
+        // close contentlet div
+        sb.append("#if($API_EDIT_MODE)")
+            .append("</div>")
+            .append("#end");
+
+        return writeOutVelocity(filePath, sb.toString());
+    }
+
+    @SuppressWarnings("unchecked")
+    public List<Contentlet> sortContentlets(List<Contentlet> contentletList, String sortBy) {
+
+        Logger.debug(this.getClass(), "I'm ordering by " + sortBy);
+
+        if (sortBy.equals("tree_order")) {
+            Collections.sort(contentletList, new WebAssetSortOrderComparator());
+        } else {
+            Collections.sort(contentletList, new ContentComparator("asc"));
+        }
+
+        return contentletList;
+
+    }
+
+    /**
+     * Will remove all contentlet files within a structure for both live and working. Uses the
+     * system user.
+     * 
+     * @param contentlets
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    public void invalidate(Structure structure) throws DotDataException, DotSecurityException {
+        ContentletAPI conAPI = APILocator.getContentletAPI();
+        int limit = 500;
+        int offset = 0;
+        List<Contentlet> contentlets = conAPI.findByStructure(structure, APILocator.getUserAPI()
+            .getSystemUser(), false, limit, offset);
+        int size = contentlets.size();
+        while (size > 0) {
+            for (Contentlet contentlet : contentlets) {
+                invalidate(contentlet);
+            }
+            offset += limit;
+            contentlets = conAPI.findByStructure(structure, APILocator.getUserAPI()
+                .getSystemUser(), false, limit, offset);
+            size = contentlets.size();
+        }
+    }
+
+
+
+    @Override
+    public InputStream writeObject(String id1, String id2, PageMode mode, String language, String filePath) {
+        try {
+
+            Identifier identifier = APILocator.getIdentifierAPI()
+                .find(id1);
+            Contentlet contentlet = null;
+            if (CacheLocator.getVeloctyResourceCache()
+                .isMiss(filePath)) {
+                if (LanguageWebAPI.canDefaultContentToDefaultLanguage()) {
+                    LanguageAPI langAPI = APILocator.getLanguageAPI();
+                    language = Long.toString(langAPI.getDefaultLanguage()
+                        .getId());
+                } else {
+                    throw new ResourceNotFoundException("Contentlet is a miss in the cache");
+                }
+            }
+
+            try {
+                contentlet = APILocator.getContentletAPI()
+                    .findContentletByIdentifier(identifier.getId(), mode.showLive, new Long(language), APILocator.getUserAPI()
+                        .getSystemUser(), true);
+            } catch (DotContentletStateException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)